### PR TITLE
AOI Mux updates

### PR DESCRIPTION
**Pull request by ankita0805**
Changes to AOI mux structure. Corresponding tests are updated.

### CODE REVIEW

**Comment by Kuree** [comment]
@ankita0805 Is the x-prop test still not working? If so, it is because of fault?

**Comment by ankita0805** [comment]
I think so. I have those tests in a separate branch to keep these structural changes separate from Xprop tests and for Lenny to review the tests: 

https://github.com/StanfordAHA/gemstone/tree/xprop

File: gemstone/common/AN2D0BWP16P90.sv [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module AN2D0BWP16P90 (
```
Comment (rsetaluri) [inline gemstone/common/AN2D0BWP16P90.sv]: why are these files included?

File: gemstone/common/mux_wrapper_aoi.py [hunk 2]
```diff
@@ -38,13 +34,18 @@ def definition(io):
...
                     f = open("mux_aoi.sv", "w")
                     num_sel = math.ceil(math.log(height, 2))
+                    num_ops = math.ceil(height / 2)
                 num_inputs = math.pow(2, num_sel)
 
-                f.write(f"module mux_aoi_{height}_{width} ( \n")
-
+                # ======= MUX MODULE =========
+                if mux_type == mux_type.Const:
+                    f.write(f"module mux_aoi_const_{height}_{width} ( \n")
```
Comment (rsetaluri) [inline gemstone/common/mux_wrapper_aoi.py]: This should be refactored to use a in-memory string rather than write to file. There is no need to write to file. We can use the function `DefineFromVerilog` rather than `DefineFromVerilogFile`

**Review by rsetaluri — CHANGES_REQUESTED** [review]

File: gemstone/common/AN2D0BWP16P90.sv (hunk 1 above)
Comment (Kuree) [reply gemstone/common/AN2D0BWP16P90.sv]: Because these are standard cell gates she needs to optimize in the PD flow. I personally think we should refactor this to allow standard cell stub in the future. However, I think `gemstone` should allow standard cell stubs to be inserted throughout the design, if `gemstone` is every going to be used for tape out.

Once your fix for the magma is done, we can try to refactor it.

Comment (ankita0805) [reply gemstone/common/AN2D0BWP16P90.sv]: Some standard cells are instantiated in the module and Fault will need them for the tests.

File: gemstone/common/mux_wrapper_aoi.py (hunk 2 above)
Comment (ankita0805) [reply gemstone/common/mux_wrapper_aoi.py]: Do you have an example?

Comment (ankita0805) [reply gemstone/common/mux_wrapper_aoi.py]: Moved these files to test/common/rtl folder based off discussion with @Kuree. Note these files are needed only for Fault tests and *not* for the design.

Comment (rsetaluri) [reply gemstone/common/mux_wrapper_aoi.py]: That makes sense for those standard cell files. I'm talking about these `f.write` statements. These should just append to a string in memory rather than writing to file then reading from that file. Just extra file IO overhead we don't need.

**Review by rsetaluri — CHANGES_REQUESTED** [review]

**Comment by Kuree** [comment]
@ankita0805 The new changes broke garnet functionalities, especially the power domain pass. I will take a look and fix it.

EDIT: I will also rewrite the file io into in memory string.

**Review by rsetaluri — APPROVED** [review]
